### ISSUE TITLE
Decoding error: out of bounds in StringDictionaryVector

### ISSUE THREAD

**Issue by springmeyer** [maplibre/maplibre-tile-spec#146]
With the latest code, in the vectorized path, we are hitting:

```
java.lang.IndexOutOfBoundsException
	at java.base/java.nio.Buffer.checkIndex(Buffer.java:743)
	at java.base/java.nio.HeapIntBuffer.get(HeapIntBuffer.java:169)
	at com.mlt.vector.dictionary.StringDictionaryVector.getValueFromBuffer(StringDictionaryVector.java:71)
	at com.mlt.vector.dictionary.StringDictionaryVector.getValueFromBuffer(StringDictionaryVector.java:12)
	at com.mlt.vector.Vector.getValue(Vector.java:29)
	at com.mlt.vector.FeatureTable$1.next(FeatureTable.java:52)
	at com.mlt.vector.FeatureTable$1.next(FeatureTable.java:31)
	at com.mlt.tools.CliUtil.decodeFeatureTables(CliUtil.java:15)
	at com.mlt.tools.Encode.main(Encode.java:259)|
```

Testcase:

```
cd java
./gradlew cli
git clone git@github.com:mapbox/mvt-fixtures.git
java -jar build/libs/encode.jar -mvt mvt-fixtures/real-world/norway/12-2174-1070.mvt -decode -vectorized
```

**Comment by springmeyer** [comment]
This still fails after #158, but with a new error:

```
java.lang.IllegalArgumentException: Specified physical level technique not yet supported: NONE
	at com.mlt.decoder.IntegerDecoder.decodeIntStream(IntegerDecoder.java:86)
	at com.mlt.decoder.GeometryDecoder.decodeGeometryColumn(GeometryDecoder.java:29)
	at com.mlt.decoder.MltDecoder.decodeMlTile(MltDecoder.java:77)
	at com.mlt.tools.Encode.main(Encode.java:269)
```

**Comment by springmeyer** [comment]
This now appears that it is a duplicate of #148

**Comment by springmeyer** [comment]
Confirmed it is a dupe of https://github.com/maplibre/maplibre-tile-spec/issues/148
